sonar-ballerina-plugin/src/main/java/io/ballerina/sonar/BallerinaRulesDefinition.java

```diff
@@ -0,0 +1,86 @@
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package io.ballerina.sonar;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonArray;
+import org.sonar.api.SonarRuntime;
+import org.sonar.api.server.rule.RulesDefinition;
+import org.sonarsource.analyzer.commons.RuleMetadataLoader;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.nio.charset.StandardCharsets;
+import java.util.ArrayList;
+import java.util.List;
+
+import static io.ballerina.sonar.Constants.JSON_PROFILE_PATH;
+import static io.ballerina.sonar.Constants.LANGUAGE_KEY;
+import static io.ballerina.sonar.Constants.RULE_KEY;
+import static io.ballerina.sonar.Constants.RULE_REPOSITORY_KEY;
+import static io.ballerina.sonar.Constants.RULE_REPOSITORY_NAME;
+import static io.ballerina.sonar.Constants.RULE_RESOURCE_FOLDER;
+
+/**
+ * Represents the implementation of the {@link RulesDefinition} class for Ballerina.
+ *
+ * @since 0.1.0
+ */
+public class BallerinaRulesDefinition implements RulesDefinition {
+    private final SonarRuntime runtime;
+    private final Gson gson = new Gson();
+
+    public BallerinaRulesDefinition(SonarRuntime runtime) {
+        this.runtime = runtime;
+    }
+
+    @Override
+    public void define(Context context) {
+        NewRepository repository = context.createRepository(RULE_REPOSITORY_KEY, LANGUAGE_KEY);
+        repository.setName(RULE_REPOSITORY_NAME);
+        RuleMetadataLoader ruleMetadataLoader = new RuleMetadataLoader(RULE_RESOURCE_FOLDER, JSON_PROFILE_PATH,
+                runtime);
+        ruleMetadataLoader.addRulesByRuleKey(repository, loadRuleKeys());
+        repository.done();
+    }
+
+    private List<String> loadRuleKeys() {
+        InputStream inputStream = BallerinaRulesDefinition.class.getResourceAsStream("/ruleKeys.json");
+        StringBuilder builder = new StringBuilder();
+        if (inputStream != null) {
+            try (
+                    InputStreamReader inputStreamReader = new InputStreamReader(inputStream, StandardCharsets.UTF_8);
+                    BufferedReader br = new BufferedReader(inputStreamReader)
+            ) {
+                String content = br.readLine();
+                builder.append(content);
+                while ((content = br.readLine()) != null) {
+                    builder.append(System.lineSeparator()).append(content);
+                }
+            } catch (IOException ex) {
+                throw new IllegalStateException(ex);
+            }
+        }
+        JsonArray rules = gson.fromJson(builder.toString(), JsonArray.class);
```

Comment:
I think we can write this in a more declarative style. WDYT?

```suggestion
        String body;
        if (inputStream != null) {
            try (
                    InputStreamReader inputStreamReader = new InputStreamReader(inputStream, StandardCharsets.UTF_8);
                    BufferedReader br = new BufferedReader(inputStreamReader)
            ) {
                body = br.lines().collect(Collectors.joining(System.lineSeparator()));
            } catch (IOException ex) {
                throw new IllegalStateException(ex);
            }
        } else {
            body = ""; // Can't we just return here without doing anything?
        }
        JsonArray rules = gson.fromJson(body, JsonArray.class);
```

Reply:
`body = ""; // Can't we just return here without doing anything?`, When parsing a string to JSON using GSON library, the String variable cannot have a null value, so this block is required.